### PR TITLE
Invalidate FileInfo after CreateText and AppendText

### LINKED ISSUE #77453: FileInfo.CreateText and FileInfo.AppendText don't invalidate cache
### Description

The `CreateText` and `AppendText` methods on the `FileInfo` class don't invalidate the cache. The `Create` method does, though. This results in an inconsistent developer experience.

### Reproduction Steps

```csharp
FileInfo fileInfoForCreateText = new("foo.txt");
Console.WriteLine("File exists before CreateText: {0}", fileInfoForCreateText.Exists);
using (StreamWriter streamWriter = fileInfoForCreateText.CreateText())
{
    Console.WriteLine("File exists after CreateText: {0}", fileInfoForCreateText.Exists);
}

FileInfo fileInfoForCreate = new("foo2.txt");
Console.WriteLine("File exists before Create: {0}", fileInfoForCreate.Exists);
using (FileStream fileStream = fileInfoForCreate.Create())
{
    Console.WriteLine("File exists after Create: {0}", fileInfoForCreate.Exists);
}
```

### Expected behavior

The output should be
```
File exists before CreateText: False
File exists after CreateText: True
File exists before Create: False
File exists after Create: True
```

### Actual behavior

The output is
```
File exists before CreateText: False
File exists after CreateText: False
File exists before Create: False
File exists after Create: True
```

### Regression?

Before 1b683ecaa0838cc8626d4124a8bfeb7a9f4610fa, there was no difference between `CreateText` and `Create` considering the invalidation of the `FileInfo` object. Even if the methods now behave differently than in ie. .NET Framework 4.x, they should behave consistently.

### Known Workarounds

You can get the correct information by creating a new instance of `FileInfo` for the same path.

### Configuration

- .NET 7.0.100-rc.2.22477.23
- Windows 10 21H1 (Build 19043.2130)
- x64
- Configuration shouldn't matter, as the [relevant code](https://github.com/dotnet/runtime/blob/v7.0.0-rc.2.22472.3/src/libraries/System.Private.CoreLib/src/System/IO/FileInfo.cs#L84-L88) is platform agnostic

### Other information

The invalidation is a new feature added in #34229.

### PR DESCRIPTION
**Pull request by hangy**
Similarly to `FileInfo.Create()`, `FileInfo.CreateText()` and `FileInfo.AppendText()` now invalidate the data cached inside a `FileInfo` instance, so that `FileInfo.Exists` returns the correct value when being called afterwards.

Fix #77453

### CODE REVIEW

File: src/libraries/System.IO.FileSystem/tests/FileInfo/CreateText.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+namespace System.IO.Tests
+{
+    public class FileInfo_CreateText_str : FileSystemTest
+    {
+        [Fact]
+        public void FileInfoInvalidAfterCreateText()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFilePath = Path.Combine(testDir.FullName, GetTestFileName());
+            FileInfo info = new FileInfo(testFilePath);
```
Comment (adamsitnik) [inline src/libraries/System.IO.FileSystem/tests/FileInfo/CreateText.cs:15]:
```suggestion
            FileInfo info = new FileInfo(GetTestFilePath());
```

File: src/libraries/System.Private.CoreLib/src/System/IO/FileInfo.cs [hunk 2]
```diff
@@ -200,5 +200,12 @@ public FileInfo Replace(string destinationFileName, string? destinationBackupFil
 
         [SupportedOSPlatform("windows")]
         public void Encrypt() => File.Encrypt(FullPath);
+
+        private StreamWriter CreateStreamWriter(bool append)
+        {
+            StreamWriter streamWriter = new StreamWriter(NormalizedPath);
```
Comment (adamsitnik) [inline src/libraries/System.Private.CoreLib/src/System/IO/FileInfo.cs:206]: we need to pass `append` argument to the `StreamWriter` ctor:

```suggestion
            StreamWriter streamWriter = new StreamWriter(NormalizedPath, append);
```

**Review by adamsitnik — COMMENTED** [review]
@hangy thank you for your contribution! Please take a look at my comments.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @hangy !

**Comment by hangy** [comment]
> @hangy thank you for your contribution! Please take a look at my comments.

Thanks for your input! Can't believe I actually missed to pass the append parameter. The test methods do look more concise, too. Hope I can provide better PRs if I get more familiar with the code base. 😃

**Comment by adamsitnik** [comment]
> Thanks for your input! Can't believe I actually missed to pass the append parameter. The test methods do look more concise, too.

Don't worry about it! I've been working on the .NET Team for almost 5 years now and I usually get at least ten comments for even simplest PRs I create.

> Hope I can provide better PRs if I get more familiar with the code base. 

Looking forward to more PRs from you!

**Pull request merged**